pkg/resources/containers/container.go

```diff
@@ -0,0 +1,89 @@
+package containers
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	//"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"os"
+)
+
+const (
+	TimeInSeconds = 5
+)
+
+// Go build-time variables
+var ImageRepository string
+
+//func MakeContainer
+func MakeContainer(customResourceName string, version string, envVarArray []corev1.EnvVar) corev1.Container {
+
+	container := corev1.Container{
+		Name:  customResourceName + "-container",
+		Image: getImageForCR(version),
+		Env:   envVarArray,
+		ReadinessProbe: &corev1.Probe{
+			InitialDelaySeconds: TimeInSeconds,
+			TimeoutSeconds:      1,
+			PeriodSeconds:       TimeInSeconds,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Port:   intstr.FromString("nginx"),
+					Path:   "/online",
+					Scheme: "HTTPS",
+				},
+			},
+		},
+		LivenessProbe: &corev1.Probe{
+			TimeoutSeconds: 1,
+			PeriodSeconds:  10,
+			Handler: corev1.Handler{
+				HTTPGet: &corev1.HTTPGetAction{
+					Port:   intstr.FromString("nginx"),
+					Path:   "/online",
+					Scheme: "HTTPS",
+				},
+			},
+		},
+		Ports: []corev1.ContainerPort{
+			{
+				Name:          "nginx",
+				ContainerPort: 8443,
+				Protocol:      "TCP",
+			},
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("900m"),
```

Comment:
Can be changed to 1 CPU.

Reply:
faced Kubernetes API conversion issue, behind the scene it should convert 1 cpu to 1000m, even if mentioned 1000m it gives an error. so for temporary fixed mentioned 900m. though it worked perfectly in the YAML file but make issue in code. https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-cpu

Reply:
What about `resource.MustParse("1")`?

Reply:
I tested before, it was not converting convert 1 to 1000m, and cause error comparison error.  need to verify if it accepts `resource.MustParse("1Gi")) ` .

Reply:
@astefanutti my fault. it works perfectly with 	`corev1.ResourceCPU:    resource.MustParse("1")`, before I was passing 	`corev1.ResourceCPU:    resource.MustParse("1.0")`  which was making the conversion issue.  pushed the latest changes.